### DISCUSSION TITLE
Can't prevet automatically redirecting/fixing url

### DISCUSSION THREAD

**Discussion by spmsupun** [vercel/next.js#29248] [Help]
### What version of Next.js are you using?

11.0

### What version of Node.js are you using?

16

### What browser are you using?

chrome

### What operating system are you using?

ubuntu/linux

### How are you deploying your application?

next start

### Describe the Bug

When you have a page (ex: /test) and if you request it will automatically remove the last slash:

- http://localhost:3000/test/  to http://localhost:3000/test/
- http://localhost:3000/test/?q=22  to http://localhost:3000/test?q=22

this is okay unless if you want to keep the exact URL you have entered. 

When I use rewrites and I have a rewrite for socket.io. in socket io the request URL should be something like this:
http://0.0.0.0:30/socket/socket.io/?EIO=4&transport=polling&t=NhhqTB4
but NextJs convert to this removing the last slash
http://0.0.0.0:30/socket/socket.io?EIO=4&transport=polling&t=NhhqTB4
which is wrong in socket.io.

UPDATE:
there's an option called [trailing-slash](https://nextjs.org/docs/api-reference/next.config.js/trailing-slash
) but it always redirects to the current URL with a slash which is not expected behavior

### Expected Behavior

Entered URL shouldn't be changed

### To Reproduce

Create any nextJs project and try routes.

**Comment by ijjk** [comment]
Hi, the trailingSlash config is currently expected to apply for all URLs to prevent duplicate content, if you are rewriting this request and the upstream server needs to not have the trailing slash you can omit it from the rewrite's destination e.g. `{ source: '/socket.io', destination: 'http://localhost:30/socket/socket.io' }`
